File: src/main/java/imaxct/dao/ICourseDao.java
package imaxct.dao;

import imaxct.domain.Course;

import java.util.List;

/**
 * Created by maxct on 2017/4/9.
 */
public interface ICourseDao {
    boolean addCourse(Course course);

    boolean deleteCourse(Course course);

    boolean updateCourse(Course course);

    List<Course> getCourseByName(String name);

    List<Course> getAllCourses();

    Course getById(int id);

    List<Course> getFromId(int id);
}

File: src/main/java/imaxct/dao/ICourseDao.kt
package imaxct.dao

import imaxct.domain.Course

/**
 * Created by maxct on 2017/4/9.
 */
interface ICourseDao {
    fun addCourse(course: Course): Boolean

    fun deleteCourse(course: Course): Boolean

    fun updateCourse(course: Course): Boolean

    fun getAllCourses(): List<Course>

    fun getCourseById(id: Int): Course

    fun getCourseFromId(id: Int): List<Course>

    fun allCourses(): List<Course>
}
